src/main/java/mate/academy/lib/BookDaoImpl.java

```diff
@@ -0,0 +1,105 @@
+package mate.academy.lib;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String URL = "jdbc:mysql://localhost:3306/book_schema?serverTimezone=UTC";
+    private static final String USER = "your_username";
+    private static final String PASSWORD = "your_password";
+
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = DriverManager.getConnection(URL, USER, PASSWORD);
+             PreparedStatement statement = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to create book: " + book, e);
```

Comment:
Ensure that the `DataProcessingException` class is defined as specified in the task description. It should extend `RuntimeException` and include a constructor with `String message` and `Throwable cause` parameters.